File: data/can_sort/Original_Solution.cpp
class Solution {
public:
    bool canSortArray(vector<int>& nums) {
        int n = nums.size();
        
        vector<int> c = nums;
        sort(begin(c),end(c));
        
        int prev=0 , i=0;
        int a = __builtin_popcount(nums[0]);
        while(i < n) {
            int b = __builtin_popcount(nums[i]);
            
            while(i<n && a==__builtin_popcount(nums[i])) {
                i++;
            }
            sort(nums.begin()+prev,nums.begin()+i);
            prev = i;
            if(i < n)   a = __builtin_popcount(nums[i]);
        }
        return c==nums;
    }
};

File: data/find_polygon/LeetCode_Solutions/Original_Solution.java
class Solution {
public long largestPerimeter(int[] nums) {
final int n = nums.length;
Arrays.sort(nums);
long result = -1;
long prefix = nums[0] + nums[1];
for (int i = 2; i < n; i++) {
if (prefix > nums[i]) {
result = Math.max(result, prefix + nums[i]);
}
prefix += nums[i];
}
return result;
}
}